library/arith.lisp

```diff
@@ -502,6 +508,159 @@ The fields are defined as follows:
       (lisp Double-Float (x y)
         (cl:coerce (cl:/ x y) 'cl:double-float)))))
 
+(coalton-toplevel
+  (define-class (Trigonometric :a)
+    (sin   (:a -> :a))
+    (cos   (:a -> :a))
+    (tan   (:a -> :a))
+    (asin  (:a -> :a))
+    (acos  (:a -> :a))
+    (atan  (:a -> :a)))
+
+  (declare sincos (Trigonometric :a => :a -> (Tuple :a :a)))
+  (define (sincos x)
+    (Tuple (sin x) (cos x)))
+
+  (define-class (Exponentiable :a)
+    (expt (:a -> :a -> :a))
+    (log  (:a -> :a -> :a)))
+
+  (declare sqrt ((Into Fraction :a) (Exponentiable :a) => :a -> :a))
+  (define (sqrt x)
+    "Returns the square root of the input"
+    (expt x (into 1/2)))
+
+  (define-class ((Num :a) (Into Fraction :a) (Dividable :a :a) (Trigonometric :a) (Exponentiable :a) => Float :a)
```

Comment:
document this class, be sure to say something about commonality / practicality, and that it actually has very little to do with the abstract definition of a float